python/sglang/srt/layers/linear.py

```diff
@@ -1316,9 +1317,8 @@ def forward(self, input_, skip_all_reduce=False):
         # Only fuse bias add into GEMM for rank 0 (this ensures that
         # bias will not get added more than once in TP>1 case)
         bias_ = None if (self.tp_rank > 0 or self.skip_bias_add) else self.bias
-        with use_symmetric_memory(parallel_state.get_tp_group()) as sm:
+        with use_symmetric_memory(get_tp_group(), disabled=not is_dp_max_padding()):
```

Comment:
you are writing super-critical runtime code, so please try to reduce python overhead as much as possible.

Can you achieve the following effect?
When symmetric_memory flag is disabled, the overhead should just be < 2 python if/else or function call per layer.

I feel the current way of inserting many `with use_symmetric_memory(get_tp_group(), disabled=not is_dp_max_padding()):` adds python overhead

Reply:
> with use_symmetric_memory(get_tp_group(), disabled=not is_dp_max_padding())
I have not noticed any cpu overhead here. 

>> I feel the current way of inserting many with use_symmetric_memory(get_tp_group(), disabled=not is_dp_max_padding()): adds python overhead

The memory pool stack of pytorch has already a lot of conditionals and checks. Symmetric memory context manager are always around memory allocations. So I don't believe adding an empty context manager with an extra conditional add overhead. I would be glad to be proven wrong.